packages/ckeditor5-emoji/src/emojipicker.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module emoji/emojipicker
+ */
+
+import { Picker } from 'emoji-picker-element';
+
+import {
+	ButtonView,
+	clickOutsideHandler,
+	ContextualBalloon,
+	View
+} from 'ckeditor5/src/ui.js';
+
+import { Plugin, type Editor } from 'ckeditor5/src/core.js';
+import EmojiLibraryIntegration from './emojilibraryintegration.js';
+
+type EmojiClickEvent = Event & {
+	detail: {
+		unicode: string;
+	};
+};
+
+/**
+ * The emoji picker plugin.
+ *
+ * Introduces the `'emoji'` dropdown.
+ */
+export default class EmojiPicker extends Plugin {
+	private _balloon!: ContextualBalloon;
+	declare private _emojiDialog: EmojiDialog;
```

Comment:
Dialog is something different (opening a separate "window"). I would call it `EmojiBalloon` or `EmojiView`?